### DISCUSSION TITLE
Pinecone api failed to work on Render flowise

### DISCUSSION THREAD

**Discussion by PingUmich** [FlowiseAI/Flowise#1595] [Q&A]
I deployed flowise to render yesterday, and chose node instead of docker.  After that, i upload chatflow I created in my localhost. But pinecone upset failed for some reason. I did checked my pinecone api key, and tried a couple times. Thank you so much! The information from log as following screenshot: 

![1706034797051](https://github.com/FlowiseAI/Flowise/assets/156003097/05bf906b-5776-4e12-8543-56e3f491e4bb)

**Comment by rvy474** [comment]
I have faced this same error when using Pinecone's serverless hosting option. What are you using?

**Comment by PingUmich** [comment]
> I have faced this same error when using Pinecone's serverless hosting option. What are you using?

![1706624396822](https://github.com/FlowiseAI/Flowise/assets/156003097/d81b96c4-007b-4f14-9d97-2a2c01a17697)
Does this mean I use Pinecone serverless hosting option? I used Pinecone as a vector store. I attached my chat flow here.  

It seems I found the reason: when I created pinecone index name and API key name, I used the same names. It did not have a problem when I run locally. I just changed the Pinecone index and API name differently, it worked in this sample chatflow above. However, when I tried on my another project, which includes 156 urls. I received "FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory" and "error Command failed with exit code 134". Do you know how to use flowise for large database by chance? Thank you so much!

**Reply by HenryHengZJ** [reply]
hmm I think the best way to handle heap memory issue is to split into smaller work, and do it multiple times.

**Comment by PingUmich** [comment]
> hmm I think the best way to handle heap memory issue is to split into smaller work, and do it multiple times.

Thank you so much for your help, Henry! Could you please give me some suggestions about following questions?

1. We have about 200 URLs as our database, is there any "smart" way to upload all the urls? I currently made about 200 url upload in my chat flow, which looks bed because too much nodes at one singer chatflow interface. The urls likes a tree that has several main urls with many branches. I used cheerio web scraper as a document loader. I used to believe the scraper can crawl the websties after I set parameters so that I did not need to upload all 200 urls. But I found if I ask questions that from the subsite I did not upload, the answers were not correct. 
2. You mentioned the best way to handle heap memory issue is to split into smaller work, and do it multiple times. I can I split, do it multiple times, then connect all of sources together?

Really appreciate your help!
